server/aws-lsp-codewhisperer/src/language-server/agenticChat/agenticChatController.ts

```diff
@@ -1140,13 +1191,14 @@ export class AgenticChatController implements ChatHandlers {
             return createAuthFollowUpResult(authFollowType)
         }
 
-        this.#log(`Q api request error ${err instanceof Error ? JSON.stringify(err) : 'unknown'}`)
-        this.#debug(`Q api request error stack ${err instanceof Error ? JSON.stringify(err.stack) : 'unknown'}`)
-        this.#debug(`Q api request error cause ${err instanceof Error ? JSON.stringify(err.cause) : 'unknown'}`)
-        return new ResponseError<ChatResult>(
-            LSPErrorCodes.RequestFailed,
-            err instanceof Error ? err.message : 'Unknown request error'
-        )
+        const backendError = err.cause
+        // Send the backend error message directly to the client to be displayed in chat.
+        return {
```

Comment:
Another, more LSP, way to handle this would be to return `ResponseError` in result of LSP request. Then extension would need to translate that to error on the UI by calling "errorMessage" command with params through postMessage.

Reply:
Yeah, that makes sense. My thinking was that if we wanted any custom formatting on this message in the future it could be done once on the LSP side rather than reimplemented by each client.

Reply:
`ResponseError` has generic `data` param, so that might be used. Custom formatting could also be on `chat-client` side ([here](https://github.com/aws/language-servers/blob/6e813fdc97c1494337cd9ffc724ecd25331625b8/chat-client/src/client/mynahUi.ts#L740)), depending on what exactly you mean.